### PR TITLE
KML fillPattern

### PR DESCRIPTION
**Pull request by danji90**
- Added kml writing capability for CanvasPatterns
- When detecting a CanvasPattern in a Fill style  fillPattern: <string> is saved in the feature properties
- This string can be used to create custom CanvasPatterns in the Client app

### CODE REVIEW

File: src/utils/KML.js [hunk 1]
```diff
@@ -306,6 +306,12 @@ const writeFeatures = (layer, featureProjection) => {
       }
     }
 
+    if (newStyle.fill && newStyle.fill.getColor() instanceof CanvasPattern) {
+      // eslint-disable-next-line no-console
```
Comment (friedjoff) [inline src/utils/KML.js:310]:
```suggestion
```

File: src/utils/KML.js [hunk 2]
```diff
@@ -306,6 +306,12 @@ const writeFeatures = (layer, featureProjection) => {
       }
     }
 
+    if (newStyle.fill && newStyle.fill.getColor() instanceof CanvasPattern) {
```
Comment (oterral) [inline src/utils/KML.js:309]: You should exclude also CanvasGradient

Comment (oterral) [reply src/utils/KML.js:309]: and test if f.get('fillPattern') exists . maybe just a :

``` 
if (newStyle.fill &&  f.get('fillPattern'))
``` 

is enough

Comment (danji90) [reply src/utils/KML.js:309]: @oterral Did you mean something like this?
`if (
      (newStyle.fill && newStyle.fill.getColor() instanceof CanvasPattern) ||
      newStyle.fill.getColor() instanceof CanvasGradient
    ) {
      if (f.get('fillPattern')) {
        clone.set('fillPattern', f.get('fillPattern'));
        newStyle.fill = null;
      }
    }`

Comment (oterral) [reply src/utils/KML.js:309]: no just :
``` 
if (newStyle.fill &&  f.get('fillPattern')) {
   clone.set('fillPattern', f.get('fillPattern'));
   newStyle.fill = null;
}
```

**Comment by oterral** [comment]
Perfect thx

**Pull request merged**